src/components/common/ContactPopupForm.tsx

```diff
@@ -0,0 +1,192 @@
+import { FC } from 'react'
+import React, { useEffect, useState } from 'react'
+import useAxios from 'axios-hooks'
+import { useRouter } from 'next/router'
+import ContactFormProps from 'types/contact-form'
+
+const ContactPopupForm: FC = () => {
+  const router = useRouter()
+  const [isFormFilled, setIsFormFilled] = useState(false);
+  const [contactObj, setContactObj] = useState<ContactFormProps>({
+    firstname: '',
+    lastname: '',
+    email: '',
+    message: '',
+    pageUri: '',
+    phone: '',
+    location: '',
+    serviceType: '',
+    dedicatedStaff: ''
+  })
+
+  useEffect(() => {
+    const { serviceType, dedicatedStaff, ...popupFormRequiredFields } = contactObj
+    setIsFormFilled(Object.values(popupFormRequiredFields).every((value) => value.trim() !== ''))
+  }, [contactObj])
+
+  useEffect(() => {
+    setContactObj(() => ({
+      ...contactObj,
+      pageUri: window.location.href
+    }))
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [contactObj.pageUri])
+
+  const handleInputChange = (
+    e: React.FormEvent<HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>,
+    prop: string
+  ) => {
+    setContactObj({
+      ...contactObj,
+      [prop]: e.currentTarget.value
+    })
+    console.log(contactObj);
```

Comment:
We should remove any console logs for Production. they are fine for development only